app/services/nomis/elite2/offender_api.rb

```diff
@@ -111,6 +118,15 @@ def self.get_image(booking_id)
 
     private
 
+      def self.get_recall_flags(offenders)
+        search_route = '/prisoner-numbers'
+        search_key = "#{search_route}_#{Digest::SHA256.hexdigest(offenders.to_s)}"
+        Rails.cache.fetch(search_key,
+                          expires_in: Rails.configuration.cache_expiry) {
+          search_client.post(search_route, prisonerNumbers: offenders)
+        }.index_by { |o| o.fetch('prisonerNumber') }
+      end
```

Comment:
Just my opinion, but this doesn't need to be followed:

It might be nice for the `get_recall_flags` method to return ONLY a hash of offender number => recall flag.

This would mean the method is doing exactly what it says on the tin – i.e. getting recall flags, and nothing else.

And it might also have the side effect of make the data merge that happens above (in the `#list` and `#get_offender` methods above) a little more obvious / easier to follow.